GitCommands/Git/GitModule.cs

```diff
@@ -408,19 +409,24 @@ public string GitCommonDirectory
         {
             get
             {
-                if (_gitCommonDirectory == null)
+                // Get a cache of the common dir
+                // Lock needed as the command is called radidly when creating the module
+                lock (_gitCommonLock)
                 {
-                    var args = new GitArgumentBuilder("rev-parse") { "--git-common-dir" };
-                    var result = _gitExecutable.Execute(args);
+                    if (_gitCommonDirectory == null)
```

Comment:
The lock is only needed if _gitCommonDirectory == null. Duplicate this check and add it before the lock? It should be also inside the lock, to make sure its not done twice.

Reply:
Also, Russkie will probably ask you to invert the if to reduce nesting..:P

Reply:
Will change

> And to confirm, this gets reset when changing repos, does it?

When changing repo, a new GitModule is created so it is updated.

Reply:
> Also, @RussKie will probably ask you to invert the if to reduce nesting..:P

Yes, it won't hurt but makes the code easier to read